### PR TITLE
Failover for the glusterfs client

### PR DESCRIPTION
**Pull request by angeldimitrov**
fail over is added in case one of the servers is down, the client will continue working with the next one.

### CODE REVIEW

File: driver.go [hunk 1]
```diff
@@ -147,9 +146,12 @@ func (d *glusterfsDriver) mountpoint(name string) string {
 }
 
 func (d *glusterfsDriver) mountVolume(name, destination string) error {
-	server := d.servers[rand.Intn(len(d.servers))]
+	var server_nodes []string
```
Comment (calavera) [inline driver.go]: can you please, follow golang's convention and use `snakeCase` names :smile:

**Comment by angeldimitrov** [comment]
@calavera The tests are failing, but this is because there are no really tests. Shoud i remove the upscript related stuff from the pull reqests or its fine?

**Comment by calavera** [comment]
Travis is failing because the file `driver.go` is not properly formatted:

```
$ test -z "$(gofmt -s -l . | tee /dev/stderr)"
driver.go
The command "test -z "$(gofmt -s -l . | tee /dev/stderr)"" exited with 1.
```

can you please move the upstart directory into a `contrib` directory? I think it's more appropriated than having it in the root. So if someone wants to add a systemd unit the contrib could be expanded to be like:

```
contrib/
    upstrart/
       ...
    systemd/
```

**Comment by angeldimitrov** [comment]
Done.

**Comment by calavera** [comment]
Thanks a lot. I merged these changes in master manually to make the history slightly more comprehensible.

https://github.com/calavera/docker-volume-glusterfs/commits/master

:tada:

**Pull request closed without merging**